public/js/reducers/sources.js

```diff
@@ -100,10 +96,19 @@ function update(state = State(), action: Action) : Record<SourcesState> {
       }
 
       return _updateText(state, action, [action.originalSource]);
+
     case "NAVIGATE":
```

Comment:
if `willNavigate` fires the `navigate` action type we should change that to will-navigate and then fire navigate separately